src/utils/hooks/useFavs.js

```diff
@@ -0,0 +1,45 @@
+import { useEffect, useState } from 'react';
```

Comment:
I think this custom hook can use a reducer to dispatch the different use cases

Reply:
done